cyipopt/tests/unit/test_deriv_errors.py

```diff
@@ -83,43 +77,26 @@ def problem_for_instance(instance):
                            cu=instance.cu)
 
 
-def test_solve_sparse(hs071_sparse_instance):
-    instance = hs071_sparse_instance
-    problem = problem_for_instance(instance)
+def ensure_status_after_solve(nlp, x0, exception, msg, status):
+    try:
+        nlp.solve(x0)
+    except exception as e:
+        assert nlp.info["status"] == status
+        print(f"{str(e) = }")
+        assert str(e) == msg
+    else:
+        assert False, f"Expected {exception = } not raised"
 
-    x, info = problem.solve(instance.x0)
-
-    assert info['status'] == 0
 
-
-def ensure_solve_status(instance, status):
+def test_solve_sparse(hs071_sparse_instance):
+    instance = hs071_sparse_instance
     problem = problem_for_instance(instance)
 
-    problem.add_option('max_iter', 50)
     x, info = problem.solve(instance.x0)
 
-    assert info['status'] == status
-
-
-def ensure_invalid_option(instance):
-    # -12: Invalid option
-    # Thrown in invalid Hessian because "hessian_approximation"
-    # is not chosen as "limited-memory"
-    ensure_solve_status(instance, -12)
+    assert info["status"] == 0
 
 
-def ensure_invalid_number(instance):
-    # -13: Invalid Number Detected
-    ensure_solve_status(instance, -13)
-
-
-def ensure_unrecoverable_exception(instance):
-    # -100: Unrecoverable Exception
-    # *Should* be returned from errors in initialization
-    ensure_solve_status(instance, -100)
```

Comment:
Removing these `ensure_unrecoverable_exception` feels like a break in backwards compatibility at first glance.